baseServer_test.go

```diff
@@ -370,8 +407,32 @@ func TestAccessAgentByID(t *testing.T) {
 func TestMessagePrint(t *testing.T) {
 	m := make([]basePlatformSOMAS.AgentGeneratorCountPair[ITestBaseAgent], 1)
 	m[0] = basePlatformSOMAS.MakeAgentGeneratorCountPair(NewTestAgent, 1)
-	server := basePlatformSOMAS.CreateServer(m, 1, 1, time.Microsecond)
+	server := basePlatformSOMAS.CreateServer(m, 1, 1, time.Second, time.Millisecond)
 	ag := NewTestAgent(server)
 	newMsg := ag.NewTestMessage()
 	newMsg.Print()
 }
+
+func (tba *TestServer) haltedMessageHandler(newMsg InfiniteLoopMessage, receiver uuid.UUID, done chan struct{}) {
+	tba.MessageHandlerLimiter(newMsg, receiver)
+	done <- struct{}{}
```

Comment:
careful with the non-blocking write. Are you sure that something is reading from the channel? This may hang / block